diff --git a/lib/angstrom.ml b/lib/angstrom.ml
index b025121..d7067d1 100644
--- a/lib/angstrom.ml
+++ b/lib/angstrom.ml
@@ -36,6 +36,11 @@ module Bigarray = struct
    * the internal Bigstring module. *)
 end

+module Expose = struct
+  module Input = Input
+  module Parser = Parser
+end
+
 type bigstring = Bigstringaf.t


diff --git a/lib/angstrom.mli b/lib/angstrom.mli
index 6f9a4c1..9fba737 100644
--- a/lib/angstrom.mli
+++ b/lib/angstrom.mli
@@ -43,8 +43,12 @@
     {!skip_while}, Angstrom makes it easy to write efficient, expressive, and
     reusable parsers suitable for high-performance applications. *)

+module Expose : sig
+  module Input = Input
+  module Parser = Parser
+end

-type +'a t
+type +'a t = 'a Expose.Parser.t
 (** A parser for values of type ['a]. *)
