diff --git a/dune b/dune
new file mode 100644
index 0000000..59aa344
--- /dev/null
+++ b/dune
@@ -0,0 +1,3 @@
+(library
+ (name opal)
+ (preprocess future_syntax))
diff --git a/opal.ml b/opal.ml
index 7f779ee..b47aabd 100644
--- a/opal.ml
+++ b/opal.ml
@@ -41,7 +41,6 @@ let parse parser input =
 (* primitives --------------------------------------------------------------- *)
 
 type 'token input = 'token LazyStream.t
-type ('token, 'result) monad = ('result * 'token input) option
 type ('token, 'result) parser = 'token input -> ('result * 'token input) option
 
 let return x input = Some(x, input)
diff --git a/opal.mli b/opal.mli
index 1de435c..ba8a9af 100644
--- a/opal.mli
+++ b/opal.mli
@@ -11,7 +11,8 @@ val implode : char list -> string
 val explode : string -> char list
 val ( % ) : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
 
-type ('token, 'result) parser
+type 'token input = 'token LazyStream.t
+type ('token, 'result) parser = 'token input -> ('result * 'token input) option
 val parse : ('token, 'a) parser -> 'token LazyStream.t -> 'a option
 
 val return : 'a -> ('token, 'a) parser
